From d5d8cdd1b2bab0b403786db824e8663b33a4e3f4 Mon Sep 17 00:00:00 2001
From: Mugunthan V N <mugunthanvnm@ti.com>
Date: Wed, 18 Jul 2012 17:07:59 +0530
Subject: [PATCH 1052/1092] drivers: net: ethernet: cpsw: enable multicast
 packet forward to host port in dual emac

In Dual EMAC mode, multicast packets are not received by the host as CPSW is
in VLAN aware mode and unknown multicast port mask0 is 0 in the VLAN entry.
Enabling the unknown multicast port mask with host port and down stream port
to the VLAN entry.

Signed-off-by: Mugunthan V N <mugunthanvnm@ti.com>
---
 drivers/net/ethernet/ti/cpsw.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ti/cpsw.c b/drivers/net/ethernet/ti/cpsw.c
index 01c722c..8ddb749 100644
--- a/drivers/net/ethernet/ti/cpsw.c
+++ b/drivers/net/ethernet/ti/cpsw.c
@@ -355,7 +355,8 @@ static inline void cpsw_add_dual_emac_mode_default_ale_entries(
 	writel(slave->port_vlan, &slave->regs->port_vlan);
 	cpsw_ale_add_vlan(priv->ale, slave->port_vlan,
 		1 << slave_port | 1 << priv->host_port, 0,
-		1 << slave_port | 1 << priv->host_port, 0);
+		1 << slave_port | 1 << priv->host_port,
+		1 << slave_port | 1 << priv->host_port);
 	cpsw_ale_vlan_add_mcast(priv->ale,
 		priv->ndev->broadcast,
 		1 << slave_port | 1 << priv->host_port,
-- 
1.7.11.2

